src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -148,23 +150,20 @@ int BulkController::open() {
     }
 
     /* Look up endpoint addresses in supported database */
-    int i;
-    for (i = 0; bulk_supported[i].vendor_id; ++i) {
-        if ((bulk_supported[i].vendor_id == m_vendorId) &&
-                (bulk_supported[i].product_id == m_productId)) {
-            m_inEndpointAddr = bulk_supported[i].in_epaddr;
-            m_outEndpointAddr = bulk_supported[i].out_epaddr;
-#if defined(__WINDOWS__) || defined(__APPLE__)
-            m_interfaceNumber = bulk_supported[i].interface_number;
-#endif
-            break;
-        }
-    }
 
-    if (bulk_supported[i].vendor_id == 0) {
+    const bulk_supported_t* device = std::find_if(
+            std::begin(bulk_supported), std::end(bulk_supported), [this](const auto& dev) {
+                return dev.vendor_id == m_vendorId && dev.product_id == m_productId;
+            });
+    if (!device) {
```

Comment:
To does that work, I think it his points after the last element and should be:
```suggestion
    if (device == std::end(bulk_supported)) {
```

Reply:
whoops, yeah... just saw that it returned a plain pointer and checked for null out of habit.

Reply:
done